### PR TITLE
Add max rss integration test

### PR DESCRIPTION
**Pull request by saschagrunert**
This adds a test to verify the maximum amount of RSS used by the server binary.

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        let status = Command::new(CLIENT_BINARY).status()?;
+        if status.success() {
+            break;
+        }
+        assert!(i != 100);
+        thread::sleep(Duration::from_millis(200));
+    }
+
+    // Retrieve the RSS
+    let contents = fs::read_to_string(format!("/proc/{}/status", pid))?;
```
Comment (saschagrunert) [inline tests/integration.rs:35]: Is this something we can trust?

File: tests/integration.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+use std::{fs, process::Command, thread, time::Duration};
+
+// The maximum amount of allowed VmRSS in Kilobytes
+const MAX_RSS_KB: u32 = 4000;
+
+// We assume that the tests run in release mode
+const SERVER_BINARY: &str = "target/release/conmon-server";
+const CLIENT_BINARY: &str = "target/release/conmon-client";
+
+#[test]
```
Comment (rphillips) [inline tests/integration.rs:15]: if we migrate to #[tokio::test] we can leverage more of tokio by using tokio::process::Command.

Comment (saschagrunert) [reply tests/integration.rs:15]: I’ll change it.

File: tests/integration.rs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    // Retrieve the RSS
+    let contents = fs::read_to_string(format!("/proc/{}/status", pid)).await?;
+    let mut rss_res = "";
+    for line in contents.lines() {
+        if line.starts_with("VmRSS:") {
+            rss_res = line
+                .split_whitespace()
+                .nth(1)
+                .context("split by whitespace")?;
```
Comment (rphillips) [inline tests/integration.rs:46]: we can break in this if statement

**Review by rphillips — APPROVED** [review]

**Review by rphillips — CHANGES_REQUESTED** [review]
forgot the break statement.

File: tests/integration.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        assert!(i != 100);
+        thread::sleep(Duration::from_millis(200));
+    }
+
+    // Retrieve the RSS
+    let contents = fs::read_to_string(format!("/proc/{}/status", pid))?;
+    let mut rss_res = "";
+    for line in contents.lines() {
+        if line.starts_with("VmRSS:") {
```
[the remark this answers is not in the document]
Comment (haircommander) [reply tests/integration.rs]: are we compiling in CI with debug or release?

[the remark this answers is not in the document]
Comment (haircommander) [reply tests/integration.rs]: > which also seems to take the storage from the linked libraries into account.

I think VmRSS is a better indicator of what we're worried about. the linked libraries should all be shared in memory, right? the constant value isn't as worrying as the amount used linearly

File: tests/integration.rs (hunk 3 above)
Comment (saschagrunert) [reply tests/integration.rs:46]: Right, added the break.

File: tests/integration.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (saschagrunert) [reply tests/integration.rs]: > are we compiling in CI with debug or release?

We run it in release mode.

> I think VmRSS is a better indicator of what we're worried about. the linked libraries should all be shared in memory, right? the constant value isn't as worrying as the amount used linearly

Good, i just added a pmap output to the CI as additional information. Still, I'm wondering why the amount of rss literally doubles in the CI.

**Review by rphillips — APPROVED** [review]

**Pull request merged**